examples/src/createReactAgent/accessThreadLevelState.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Access Thread Level State in Message Generation
+ *
+ * This enables the agent to access and utilize the current conversation history and thread-specific state
```

Comment:
the key idea here is that the "state" of the agent may affect the system prompt/messages. eg the agent goes and fetches some data info, which is stored in state, which can then be injected into system prompt